core/arch/arm/include/kernel/thread.h

```diff
@@ -308,6 +323,9 @@ void thread_restore_irq(void);
 #define THREAD_EXCP_ABT	(1 << 2)
 #define THREAD_EXCP_ALL	(THREAD_EXCP_FIQ | THREAD_EXCP_IRQ | THREAD_EXCP_ABT)
 
+/* Defines the bit mask for the foreign interrupts */
+#define THREAD_FOREIGN_INTR	THREAD_EXCP_IRQ
```

Comment:
I'm sorry, I think we misunderstand each other. My point is that there's only a single place where `THREAD_EXCP_IRQ` would be used and that's to define `THREAD_EXCP_FOREIGN_INTR` or `THREAD_EXCP_NATIVE_INTR` depending on `CFG_GICV3`.

So by bother defining `THREAD_EXCP_IRQ` at all?